src/java/CaseChangingCharStream.java

```diff
@@ -0,0 +1,110 @@
+/*
+ * [The "BSD 3-clause license"]
```

Comment:
Are we allowed to integrate BSD 3-clause with an Eclipse Public License project?

Reply:
I'm not a lawyer but I assume the answer is yes, given that it's antlr's own license (CaseInsensitiveInputStream  as [currently included](https://github.com/aphyr/clj-antlr/blob/master/src/java/CaseInsensitiveInputStream.java) is licensed with almost the same license), and the difference mainly looks to be due to an upstream licensing change to [use standard BSD wording](https://github.com/antlr/antlr4/commits/master/LICENSE.txt). The difference between what's currently attributed in `clj-antlr` looks to just be the author string -- I suppose it's worth modifying the README as well to reflect that change.

Reply:
Yeah, I've been digging around a bit and I feel ... even more confused than when I started, haha. I'm never exactly sure how this stuff interacts with projects/modules/etc. Do we need to... modify the EPL license to specify that you have to preserve the BSD 3-clause license here as well?

Reply:
(yeah, I was clearly less of a license stickler when I started this project, haha. I actually *know* Terence, maybe I should just ask him what he thinks!)

Reply:
I don't think we can (or should!) be modifying the EPL license itself for the existing EPL code, but there's no harm in being extra cautious in complying with BSD's attribution requirements too. How about this:

1. Maintain the BSD licenses in the source of both java files in their entirety, clarifying that those two pieces of code are licensed under those respective licenses, which also means they're allowed to be in this EPL project as long as they are attributed according to the terms of their licenses. Since the library is only including the classes and not actually modifying the code with new material, there's nothing to relicense and the BSD licenses can stand as is on those classes.
2. For completion, add more explicit notes in the license section of the README formally noting that these two java components have been taken from their two respective sources, and re-include each of their BSD licenses in full there, after stating that `clj-antlr` as a whole is EPL.
3. There could conceivably also be a LICENSE file added to the repository, which both includes the EPL text in full, with text at the bottom similarly attributing these two java files and reproducing their BSD licenses in full.
4. If/when `CaseInsensitiveInputStream` gets removed from this library, the notes related to it in the README and/or LICENSE could get dropped as well.

Reply:
Yeah, I think this is probably the safest bet. I'm gonna go ahead and merge this for now, and make those changes. I also asked Terence what he thinks, so if I hear back that this is a bad approach, I'll change that. :)

Reply:
Oh! Derp, this is actually *already* in the README, under the license section. OK cool!